app/bundles/ReportBundle/Model/ReportModel.php

```diff
@@ -363,80 +363,56 @@ public function getGraphList($context = 'all')
-                                    //set the header
-                                    $header[] = $k;
-                                }
-
-                                $row[] = $formatter->_($v, $reportData['columns'][$reportData['dataColumns'][$k]]['type'], true);
-                            }
-
-                            if ($count === 0) {
-                                //write the column names row
-                                fputcsv($handle, $header);
-                            }
-
-                            //write the row
-                            fputcsv($handle, $row);
-
-                            //free memory
-                            unset($row, $reportData['data'][$count]);
+                //build the data rows
+                foreach ($reportData['data'] as $count => $data) {
+                    $row = [];
+                    foreach ($data as $k => $v) {
+                        if ($count === 0) {
+                            //set the header
+                            $header[] = $k;
                         }
 
-                        fclose($handle);
+                        $row[] = $formatter->_($v, $reportData['columns'][$reportData['dataColumns'][$k]]['type'], true);
                     }
-                );
-
-                $response->headers->set('Content-Type', 'application/force-download');
-                $response->headers->set('Content-Type', 'application/octet-stream');
-                $response->headers->set('Content-Disposition', 'attachment; filename="'.$name.'.csv"');
-                $response->headers->set('Expires', 0);
-                $response->headers->set('Cache-Control', 'must-revalidate');
-                $response->headers->set('Pragma', 'public');
 
-                return $response;
+                    if ($page === 1 && $count === 0) {
+                        fputcsv($handle, $header);
+                    }
+                    fputcsv($handle, $row);
+                }
+                return;
             case 'html':
                 $content = $this->templatingHelper->getTemplating()->renderResponse(
                     'MauticReportBundle:Report:export.html.php',
                     [
-                        'data'      => $reportData['data'],
-                        'columns'   => $reportData['columns'],
+                        'data' => $reportData['data'],
+                        'columns' => $reportData['columns'],
                         'pageTitle' => $name,
-                        'graphs'    => $reportData['graphs'],
-                        'report'    => $report,
-                        'dateFrom'  => $reportData['dateFrom'],
-                        'dateTo'    => $reportData['dateTo'],
+                        'graphs' => $reportData['graphs'],
+                        'report' => $report,
+                        'dateFrom' => $reportData['dateFrom'],
+                        'dateTo' => $reportData['dateTo'],
```

Comment:
Please return the indentation back here and on other places. Also please revert the changes from `'.'` to `' . '`